crates/egui/src/widgets/plot/items/mod.rs

```diff
@@ -1569,6 +1569,421 @@ impl PlotItem for BoxPlot {
+    rows: usize,
+    /// minimum value in colormap.
+    /// Everything smaller will be mapped to palette[0]
+    min: f64,
+    /// maximum value in heatmap
+    /// Everything greater will be mapped to palette.last()
+    max: f64,
+    /// formatter for labels on tiles
+    formatter: Box<dyn Fn(f64) -> String>,
+    /// custom mapping of values to color
+    custom_mapping: Option<Box<dyn Fn(f64) -> Color32>>,
+    /// show labels on tiles
+    show_labels: bool,
+    /// possible colors, sorted by index
+    palette: [Color32; RESOLUTION],
+    /// is widget is highlighted
+    highlight: bool,
+    /// plot name
+    name: String,
+    /// Size of one tile in plot coordinates
+    tile_size: Vec2,
+}
+
+impl<const RESOLUTION: usize> PartialEq for Heatmap<RESOLUTION> {
+    /// manual implementation of PartialEq because formatter and color mapping do not impl PartialEq.
+    ///
+    /// > NOTE: custom_mapping and formatter are ignored
+    fn eq(&self, other: &Self) -> bool {
+        self.pos == other.pos
+            && self.values == other.values
+            && self.cols == other.cols
+            && self.rows == other.rows
+            && self.min == other.min
+            && self.max == other.max
+            && self.show_labels == other.show_labels
+            && self.palette == other.palette
+            && self.highlight == other.highlight
+            && self.name == other.name
+            && self.tile_size == other.tile_size
+    }
+}
+
+impl<const RESOLUTION: usize> Heatmap<RESOLUTION> {
+    /// Create a 2D heatmap. Will automatically infer number of rows.
+    ///
+    /// - `values` contains magnitude of each tile. The alignment is row by row.
+    /// - `cols` is the number of columns (i.e. the length of each row) and must not be zero.
+    /// - `values.len()` must be a multiple of `cols`.
+    ///
+    /// Example: To display this
+    ///
+    /// | -- | -- |
+    /// | 0.0 | 0.1 |
+    /// | 0.3 | 0.4 |
+    ///
+    /// pass `values = vec![0.0, 0.1, 0.3, 0.4]` and `cols = 2`.
+    ///
+    /// Returns error type if provided parameters are inconsistent
+    pub fn new(values: Vec<f64>, cols: usize) -> Result<Self, HeatmapErr> {
+        if cols == 0 {
+            return Err(HeatmapErr::ZeroColumns);
+        }
+        if (values.len() % cols) != 0 {
+            return Err(HeatmapErr::BadLength);
+        }
+        let rows = values.len() / cols;
+        if rows == 0 {
+            return Err(HeatmapErr::ZeroRows);
+        }
+
+        // determine range
+        let mut min = f64::MAX;
+        let mut max = f64::MIN;
+        for v in &values {
+            if v < &min {
+                min = *v;
+            }
+            if v > &max {
+                max = *v;
+            }
+        }
+
+        // calculate palette
+        let base_colors = vec![
+            Color32::BLACK,
+            Color32::DARK_RED,
+            Color32::YELLOW,
+            Color32::BLACK,
+        ];
```

Comment:
it would be nice to use a standard colormap here, like [turbo](https://ai.googleblog.com/2019/08/turbo-improved-rainbow-colormap-for.html)